command/format.go

```diff
@@ -19,29 +19,55 @@ const (
 	hopeDelim = "♨"
 )
 
+type FormatOptions struct {
+	Format string
+}
+
 func OutputSecret(ui cli.Ui, format string, secret *api.Secret) int {
 	return outputWithFormat(ui, format, secret, secret)
 }
 
-func OutputList(ui cli.Ui, format string, secret *api.Secret) int {
-	return outputWithFormat(ui, format, secret, secret.Data["keys"])
+func OutputList(ui cli.Ui, format string, data interface{}) int {
+	switch data.(type) {
+	case *api.Secret:
+		secret := data.(*api.Secret)
+		return outputWithFormat(ui, format, secret, secret.Data["keys"])
+	default:
+		return outputWithFormat(ui, format, nil, data)
+	}
 }
 
-func outputWithFormat(ui cli.Ui, format string, secret *api.Secret, data interface{}) int {
-	// If we had a colored UI, pull out the nested ui so we don't add escape
-	// sequences for outputting json, etc.
-	colorUI, ok := ui.(*cli.ColoredUi)
-	if ok {
-		ui = colorUI.Ui
+// OutputWithFormat supports printing arbitrary data under a specified format.
+func OutputWithFormat(ui cli.Ui, format string, data interface{}) int {
+	switch data.(type) {
+	case *api.Secret:
+		secret := data.(*api.Secret)
+		// Best-guess effort that is this a list, so parse out the keys
```

Comment:
What's the reason for doing this? There could very well be "keys" coming back from various different backends, including k/v.

Reply:
Right, so in that case you output things that are inside `secret.Data[“keys”]`

Reply:
`OutputWithFormat` is basically the combined functionality of `OutputSecret` and `OutputList`

Reply:
> Right, so in that case you output things that are inside secret.Data[“keys”]

But....why?  :-)

Reply:
This is the original OutputList behavior: https://github.com/hashicorp/vault/pull/3897/files#diff-3e4377911a05a6f261016d1d5e1fec91L27

If you just passed in the `secret`, listing keys would be not formatted correctly:
```
Key                 Value
---                 -----
keys               [foo]
```

Instead of: 
```
$ vault list secret/
Keys
----
foo
```